### PR TITLE
#10987 [SurvNet] Filter 'only cases changed since last shared with reporting tool' does not work

### LINKED ISSUE #10987: [SurvNet] Filter 'only cases changed since last shared with reporting tool' does not work
### Bug Description

Checking the checkbox 'only cases changed since last shared with reporting tool' and apply the filter leads in an error message.
This filter is very important for the health departments because of reporting requirements.
### Steps to Reproduce

1.check the checkbox _Only cases changed since last shared with reporting tool_
2.click the button _Apply Filters_

### Expected Behavior

Filter should work and cases should be shown.
### Screenshots
Hotfix 1 76 8

image

### System Details

* Device:
* SORMAS version: 1.76.7
* Android version/Browser:
* Server URL: 1402 and release-sormas-x
* User Role: admin

### Additional Information

[log.txt](https://github.com/hzi-braunschweig/SORMAS-Project/files/10066329/log.txt)
<details>
<summary>Relevant StackTrace</summary>

```
Caused by: org.omg.CORBA.TRANSACTION_ROLLEDBACK: ----------BEGIN server-side stack trace----------org.omg.CORBA.TRANSACTION_ROLLEDBACK:   vmcid: 0x2000  minor code: 1806 completed: Maybe
		at org.glassfish.enterprise.iiop.impl.POAProtocolMgr.mapException(POAProtocolMgr.java:356)
...
		at com.sun.corba.ee.impl.presentation.rmi.codegen.CodegenStubBase.invoke(CodegenStubBase.java:226)
		at de.symeda.sormas.api.caze.__CaseFacade_Remote_DynamicStub.count(de/symeda/sormas/api/caze/__CaseFacade_Remote_DynamicStub.java)
		at de.symeda.sormas.api.caze._CaseFacade_Wrapper.count(de/symeda/sormas/api/caze/_CaseFacade_Wrapper.java)
		at de.symeda.sormas.ui.caze.AbstractCaseGrid.lambda$setLazyDataProvider$b88b9d74$1(AbstractCaseGrid.java:306)
		at com.vaadin.data.provider.CallbackDataProvider.sizeInBackEnd(CallbackDataProvider.java:142)
		at com.vaadin.data.provider.AbstractBackEndDataProvider.size(AbstractBackEndDataProvider.java:66)
		at com.vaadin.data.provider.DataProviderWrapper.size(DataProviderWrapper.java:85)
		at de.symeda.sormas.ui.utils.FilteredGrid.getItemCount(FilteredGrid.java:132)
		at de.symeda.sormas.ui.caze.CasesView.updateStatusButtons(CasesView.java:858)
		at de.symeda.sormas.ui.caze.CasesView.lambda$new$7a32c94f$2(CasesView.java:197)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:706)
		at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:399)
		at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:363)
		at com.vaadin.data.provider.AbstractDataProvider.fireEvent(AbstractDataProvider.java:90)
		at com.vaadin.data.provider.AbstractDataProvider.refreshAll(AbstractDataProvider.java:51)
		at com.vaadin.data.provider.DataProviderWrapper.refreshAll(DataProviderWrapper.java:64)
		at de.symeda.sormas.ui.caze.AbstractCaseGrid.reload(AbstractCaseGrid.java:292)
		at de.symeda.sormas.ui.caze.CasesView.lambda$createFilterBar$79546433$2(CasesView.java:571)
...
Caused by: javax.persistence.PersistenceException: org.hibernate.exception.DataException: could not execute query
		at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:154)
		at org.hibernate.query.internal.AbstractProducedQuery.list(AbstractProducedQuery.java:1626)
		at org.hibernate.query.internal.AbstractProducedQuery.getSingleResult(AbstractProducedQuery.java:1665)
		at org.hibernate.query.criteria.internal.compile.CriteriaQueryTypeQueryAdapter.getSingleResult(CriteriaQueryTypeQueryAdapter.java:111)
		at de.symeda.sormas.backend.caze.CaseFacadeEjb.count(CaseFacadeEjb.java:567)
		at de.symeda.sormas.backend.caze.CaseFacadeEjb.count(CaseFacadeEjb.java:536)
		at de.symeda.sormas.backend.caze.CaseFacadeEjb.count(CaseFacadeEjb.java:355)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
...
		at com.sun.ejb.containers.EJBObjectInvocationHandler.invoke(EJBObjectInvocationHandler.java:205)	... 129 more
Caused by: org.hibernate.exception.DataException: could not execute query
		at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:115)
		at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:37)
		at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
		at org.hibernate.loader.Loader.doList(Loader.java:2871)
		at org.hibernate.loader.Loader.doList(Loader.java:2850)
		at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2682)
		at org.hibernate.loader.Loader.list(Loader.java:2677)
		at org.hibernate.loader.hql.QueryLoader.list(QueryLoader.java:540)
		at org.hibernate.hql.internal.ast.QueryTranslatorImpl.list(QueryTranslatorImpl.java:400)
		at org.hibernate.engine.query.spi.HQLQueryPlan.performList(HQLQueryPlan.java:219)
		at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1454)
		at org.hibernate.query.internal.AbstractProducedQuery.doList(AbstractProducedQuery.java:1649)
		at org.hibernate.query.internal.AbstractProducedQuery.list(AbstractProducedQuery.java:1617)	... 191 more
Caused by: org.postgresql.util.PSQLException: Der Spaltenindex 3 ist außerhalb des gültigen Bereichs. Anzahl Spalten: 2.
		at org.postgresql.core.v3.SimpleParameterList.bind(SimpleParameterList.java:70)
		at org.postgresql.core.v3.SimpleParameterList.setLiteralParameter(SimpleParameterList.java:129)
		at org.postgresql.jdbc.PgPreparedStatement.bindLiteral(PgPreparedStatement.java:1060)
		at org.postgresql.jdbc.PgPreparedStatement.setBoolean(PgPreparedStatement.java:291)
		at jdk.internal.reflect.GeneratedMethodAccessor1061.invoke(Unknown Source)
```
</details>

Connecting error:
Bug description:
Checking the checkbox _only cases changed since last shared with reporting tool_ and apply the filter leads in an error message. If you leave the case directory afterwards without unchecking the box _only cases changed since last shared with reporting tool_ leads in an error message if you want to open the case directory again. 
Hotfix 1 76 8 1

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #10987

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/externalsurveillancetool/ExternalSurveillanceToolGatewayFacadeEjb.java [hunk 1]
```diff
@@ -86,14 +85,15 @@ public void sendEvents(List<String> eventUuids, boolean archived) throws Externa
...
 
-		Response response = ClientBuilder.newBuilder()
-			.connectTimeout(30, TimeUnit.SECONDS)
-			.build()
-			.target(serviceUrl)
-			.path("export")
-			.request()
-			.post(Entity.json(params));
-		int status = response.getStatus();
+//		Response response = ClientBuilder.newBuilder()
```
Comment (StefanKock) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/externalsurveillancetool/ExternalSurveillanceToolGatewayFacadeEjb.java:88]: Why are these changes in the test (commented out code)?

**Review by StefanKock — CHANGES_REQUESTED** [review]
1. The commit message does not state what is fixed.
2. I don't get why the `Subquery<Symptoms> symptomsSq` is now gone. Was this irrelevant?

File: sormas-backend/src/main/java/de/symeda/sormas/backend/externalsurveillancetool/ExternalSurveillanceToolGatewayFacadeEjb.java (hunk 1 above)
Comment (leventegal-she) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/externalsurveillancetool/ExternalSurveillanceToolGatewayFacadeEjb.java]: committed by mistake

**Comment by leventegal-she** [comment]
> 1. The commit message does not state what is fixed.
> 2. I don't get why the `Subquery<Symptoms> symptomsSq` is now gone. Was this irrelevant?

1. I'll rewrite the message when doing the merge
2. It was not used

**Comment by StefanKock** [comment]
I compared the before/after SQL with . Notable differences:

1. Replaced two parameter function `timestamp_subtract_days` with one parameter function `timestamp_subtract_14_days`.
2. One duplicated join `case0_.person_id` avoided (performance improvement).

<details>
<summary>Full SQL</summary>

```sql
-- BEFORE

SELECT DISTINCT case0_.id AS col_0_0_,
                case0_.uuid AS col_1_0_,
                case0_.epidNumber AS col_2_0_,
                case0_.externalID AS col_3_0_,
                case0_.externalToken AS col_4_0_,
                case0_.internalToken AS col_5_0_,
                person1_.uuid AS col_6_0_,
                person1_.firstName AS col_7_0_,
                person1_.lastName AS col_8_0_,
                case0_.disease AS col_9_0_,
                case0_.diseaseVariant AS col_10_0_,
                case0_.diseaseDetails AS col_11_0_,
                case0_.caseClassification AS col_12_0_,
                case0_.investigationStatus AS col_13_0_,
                person1_.presentCondition AS col_14_0_,
                case0_.reportDate AS col_15_0_,
                case0_.creationDate AS col_16_0_,
                region3_.uuid AS col_17_0_,
                district4_.uuid AS col_18_0_,
                facility5_.uuid AS col_19_0_,
                facility5_.name AS col_20_0_,
                case0_.healthFacilityDetails AS col_21_0_,
                pointofent6_.uuid AS col_22_0_,
                pointofent6_.name AS col_23_0_,
                case0_.pointOfEntryDetails AS col_24_0_,
                user7_.uuid AS col_25_0_,
                case0_.outcome AS col_26_0_,
                person1_.approximateAge AS col_27_0_,
                person1_.approximateAgeType AS col_28_0_,
                person1_.birthdate_dd AS col_29_0_,
                person1_.birthdate_mm AS col_30_0_,
                person1_.birthdate_yyyy AS col_31_0_,
                person1_.sex AS col_32_0_,
                case0_.quarantineTo AS col_33_0_,
                case0_.completeness AS col_34_0_,
                case0_.followUpStatus AS col_35_0_,
                case0_.followUpUntil AS col_36_0_,
                person1_.symptomJournalStatus AS col_37_0_,
                case0_.vaccinationStatus AS col_38_0_,
                case0_.changeDate AS col_39_0_,
                facility5_.id AS col_40_0_,
                region8_.uuid AS col_41_0_,
                district9_.uuid AS col_42_0_,
                district9_.name AS col_43_0_,
                CASE
                    WHEN (case0_.reportingUser_id IS NOT NULL)
                         AND case0_.reportingUser_id=27
                         OR 1=1 THEN TRUE
                    ELSE FALSE
                END AS col_44_0_,

  (SELECT
     (SELECT count(visits41_.caze_id)
      FROM Visit visits41_
      WHERE case38_.id = visits41_.caze_id)
   FROM cases case38_
   WHERE case38_.id=case0_.id ) AS col_45_0_,
                greatest(case0_.changeDate, person1_.changeDate) AS col_46_0_
FROM cases case0_
LEFT OUTER JOIN Person person1_ ON case0_.person_id=person1_.id
LEFT OUTER JOIN LOCATION location2_ ON person1_.address_id=location2_.id
LEFT OUTER JOIN Region region3_ ON case0_.region_id=region3_.id
LEFT OUTER JOIN District district4_ ON case0_.district_id=district4_.id
LEFT OUTER JOIN Facility facility5_ ON case0_.healthFacility_id=facility5_.id
LEFT OUTER JOIN PointOfEntry pointofent6_ ON case0_.pointOfEntry_id=pointofent6_.id
LEFT OUTER JOIN users user7_ ON case0_.surveillanceOfficer_id=user7_.id
LEFT OUTER JOIN Region region8_ ON case0_.responsibleRegion_id=region8_.id
LEFT OUTER JOIN District district9_ ON case0_.responsibleDistrict_id=district9_.id
LEFT OUTER JOIN users user10_ ON case0_.reportingUser_id=user10_.id
LEFT OUTER JOIN Hospitalization hospitaliz11_ ON case0_.hospitalization_id=hospitaliz11_.id
LEFT OUTER JOIN previoushospitalization previousho12_ ON hospitaliz11_.id=previousho12_.hospitalization_id
LEFT OUTER JOIN ClinicalCourse clinicalco13_ ON case0_.clinicalCourse_id=clinicalco13_.id
LEFT OUTER JOIN Symptoms symptoms14_ ON case0_.symptoms_id=symptoms14_.id
LEFT OUTER JOIN Therapy therapy15_ ON case0_.therapy_id=therapy15_.id
LEFT OUTER JOIN HealthConditions healthcond16_ ON case0_.healthConditions_id=healthcond16_.id
LEFT OUTER JOIN MaternalHistory maternalhi17_ ON case0_.maternalHistory_id=maternalhi17_.id
LEFT OUTER JOIN PortHealthInfo porthealth18_ ON case0_.portHealthInfo_id=porthealth18_.id
LEFT OUTER JOIN sormastosormasorigininfo sormastoso19_ ON case0_.sormasToSormasOriginInfo_id=sormastoso19_.id
LEFT OUTER JOIN sormastosormasshareinfo sormastoso20_ ON case0_.id=sormastoso20_.caze_id
LEFT OUTER JOIN EpiData epidata21_ ON case0_.epiData_id=epidata21_.id
LEFT OUTER JOIN exposures exposures22_ ON epidata21_.id=exposures22_.epiData_id
LEFT OUTER JOIN LOCATION location23_ ON exposures22_.location_id=location23_.id
LEFT OUTER JOIN activityascase activities24_ ON epidata21_.id=activities24_.epiData_id
LEFT OUTER JOIN LOCATION location25_ ON activities24_.location_id=location25_.id
LEFT OUTER JOIN samples samples26_ ON case0_.id=samples26_.associatedCase_id
LEFT OUTER JOIN PathogenTest pathogente27_ ON samples26_.id=pathogente27_.sample_id
LEFT OUTER JOIN Person person28_ ON case0_.person_id=person28_.id
LEFT OUTER JOIN LOCATION location29_ ON person28_.address_id=location29_.id
LEFT OUTER JOIN Visit visits30_ ON case0_.id=visits30_.caze_id
LEFT OUTER JOIN surveillancereports surveillan31_ ON case0_.id=surveillan31_.caze_id
LEFT OUTER JOIN Person person32_ ON case0_.person_id=person32_.id
LEFT OUTER JOIN immunization immunizati33_ ON person32_.id=immunizati33_.person_id
LEFT OUTER JOIN Vaccination vaccinatio34_ ON immunizati33_.id=vaccinatio34_.immunization_id
AND (
       (SELECT symptoms35_.onsetDate
        FROM Symptoms symptoms35_
        WHERE symptoms35_.id=case0_.symptoms_id )>at_end_of_day(CASE
                                                                    WHEN vaccinatio34_.vaccinationDate IS NULL THEN timestamp_subtract_days(vaccinatio34_.reportDate, ?)
                                                                    ELSE vaccinatio34_.vaccinationDate
                                                                END)
     OR (
           (SELECT symptoms36_.onsetDate
            FROM Symptoms symptoms36_
            WHERE symptoms36_.id=case0_.symptoms_id ) IS NULL)
     AND case0_.reportDate>at_end_of_day(CASE
                                             WHEN vaccinatio34_.vaccinationDate IS NULL THEN timestamp_subtract_days(vaccinatio34_.reportDate, ?)
                                             ELSE vaccinatio34_.vaccinationDate
                                         END))
LEFT OUTER JOIN immunization immunizati37_ ON vaccinatio34_.immunization_id=immunizati37_.id
WHERE case0_.deleted=?
  AND (EXISTS
         (SELECT case40_.id
          FROM externalshareinfo externalsh39_
          LEFT OUTER JOIN cases case40_ ON externalsh39_.caze_id=case40_.id
          WHERE case40_.id=case0_.id
          GROUP BY case40_.id
          HAVING case0_.changeDate>max(externalsh39_.creationDate)
          AND (case0_.changeDate IS NOT NULL)
          OR symptoms14_.changeDate>max(externalsh39_.creationDate)
          AND (symptoms14_.changeDate IS NOT NULL)
          OR hospitaliz11_.changeDate>max(externalsh39_.creationDate)
          AND (hospitaliz11_.changeDate IS NOT NULL)
          OR previousho12_.changeDate>max(externalsh39_.creationDate)
          AND (previousho12_.changeDate IS NOT NULL)
          OR therapy15_.changeDate>max(externalsh39_.creationDate)
          AND (therapy15_.changeDate IS NOT NULL)
          OR clinicalco13_.changeDate>max(externalsh39_.creationDate)
          AND (clinicalco13_.changeDate IS NOT NULL)
          OR healthcond16_.changeDate>max(externalsh39_.creationDate)
          AND (healthcond16_.changeDate IS NOT NULL)
          OR maternalhi17_.changeDate>max(externalsh39_.creationDate)
          AND (maternalhi17_.changeDate IS NOT NULL)
          OR porthealth18_.changeDate>max(externalsh39_.creationDate)
          AND (porthealth18_.changeDate IS NOT NULL)
          OR sormastoso19_.changeDate>max(externalsh39_.creationDate)
          AND (sormastoso19_.changeDate IS NOT NULL)
          OR sormastoso20_.changeDate>max(externalsh39_.creationDate)
          AND (sormastoso20_.changeDate IS NOT NULL)
          OR epidata21_.changeDate>max(externalsh39_.creationDate)
          AND (epidata21_.changeDate IS NOT NULL)
          OR exposures22_.changeDate>max(externalsh39_.creationDate)
          AND (exposures22_.changeDate IS NOT NULL)
          OR location23_.changeDate>max(externalsh39_.creationDate)
          AND (location23_.changeDate IS NOT NULL)
          OR activities24_.changeDate>max(externalsh39_.creationDate)
          AND (activities24_.changeDate IS NOT NULL)
          OR location25_.changeDate>max(externalsh39_.creationDate)
          AND (location25_.changeDate IS NOT NULL)
          OR samples26_.changeDate>max(externalsh39_.creationDate)
          AND (samples26_.changeDate IS NOT NULL)
          OR pathogente27_.changeDate>max(externalsh39_.creationDate)
          AND (pathogente27_.changeDate IS NOT NULL)
          OR person28_.changeDate>max(externalsh39_.creationDate)
          AND (person28_.changeDate IS NOT NULL)
          OR location29_.changeDate>max(externalsh39_.creationDate)
          AND (location29_.changeDate IS NOT NULL)
          OR visits30_.changeDate>max(externalsh39_.creationDate)
          AND (visits30_.changeDate IS NOT NULL)
          OR surveillan31_.changeDate>max(externalsh39_.creationDate)
          AND (surveillan31_.changeDate IS NOT NULL)
          OR vaccinatio34_.changeDate>max(externalsh39_.creationDate)
          AND (vaccinatio34_.changeDate IS NOT NULL)
          OR immunizati37_.changeDate>max(externalsh39_.creationDate)
          AND (immunizati37_.changeDate IS NOT NULL)))
  AND case0_.dontShareWithReportingTool=FALSE
ORDER BY greatest(case0_.changeDate, person1_.changeDate) DESC
LIMIT ?
```

```sql
-- AFTER

SELECT DISTINCT case0_.id AS col_0_0_,
                case0_.uuid AS col_1_0_,
                case0_.epidNumber AS col_2_0_,
                case0_.externalID AS col_3_0_,
                case0_.externalToken AS col_4_0_,
                case0_.internalToken AS col_5_0_,
                person1_.uuid AS col_6_0_,
                person1_.firstName AS col_7_0_,
                person1_.lastName AS col_8_0_,
                case0_.disease AS col_9_0_,
                case0_.diseaseVariant AS col_10_0_,
                case0_.diseaseDetails AS col_11_0_,
                case0_.caseClassification AS col_12_0_,
                case0_.investigationStatus AS col_13_0_,
                person1_.presentCondition AS col_14_0_,
                case0_.reportDate AS col_15_0_,
                case0_.creationDate AS col_16_0_,
                region8_.uuid AS col_17_0_,
                district9_.uuid AS col_18_0_,
                facility10_.uuid AS col_19_0_,
                facility10_.name AS col_20_0_,
                case0_.healthFacilityDetails AS col_21_0_,
                pointofent11_.uuid AS col_22_0_,
                pointofent11_.name AS col_23_0_,
                case0_.pointOfEntryDetails AS col_24_0_,
                user12_.uuid AS col_25_0_,
                case0_.outcome AS col_26_0_,
                person1_.approximateAge AS col_27_0_,
                person1_.approximateAgeType AS col_28_0_,
                person1_.birthdate_dd AS col_29_0_,
                person1_.birthdate_mm AS col_30_0_,
                person1_.birthdate_yyyy AS col_31_0_,
                person1_.sex AS col_32_0_,
                case0_.quarantineTo AS col_33_0_,
                case0_.completeness AS col_34_0_,
                case0_.followUpStatus AS col_35_0_,
                case0_.followUpUntil AS col_36_0_,
                person1_.symptomJournalStatus AS col_37_0_,
                case0_.vaccinationStatus AS col_38_0_,
                case0_.changeDate AS col_39_0_,
                facility10_.id AS col_40_0_,
                region13_.uuid AS col_41_0_,
                district14_.uuid AS col_42_0_,
                district14_.name AS col_43_0_,
                CASE
                    WHEN (case0_.reportingUser_id IS NOT NULL)
                         AND case0_.reportingUser_id=27
                         OR 1=1 THEN TRUE
                    ELSE FALSE
                END AS col_44_0_,

  (SELECT
     (SELECT count(visits40_.caze_id)
      FROM Visit visits40_
      WHERE case37_.id = visits40_.caze_id)
   FROM cases case37_
   WHERE case37_.id=case0_.id ) AS col_45_0_,
                greatest(case0_.changeDate, person1_.changeDate) AS col_46_0_
FROM cases case0_
LEFT OUTER JOIN Person person1_ ON case0_.person_id=person1_.id
LEFT OUTER JOIN LOCATION location2_ ON person1_.address_id=location2_.id
LEFT OUTER JOIN immunization immunizati3_ ON person1_.id=immunizati3_.person_id
LEFT OUTER JOIN Vaccination vaccinatio4_ ON immunizati3_.id=vaccinatio4_.immunization_id
AND (
       (SELECT symptoms5_.onsetDate
        FROM Symptoms symptoms5_
        WHERE symptoms5_.id=case0_.symptoms_id )>at_end_of_day(CASE
                                                                   WHEN vaccinatio4_.vaccinationDate IS NULL THEN timestamp_subtract_14_days(vaccinatio4_.reportDate)
                                                                   ELSE vaccinatio4_.vaccinationDate
                                                               END)
     OR (
           (SELECT symptoms6_.onsetDate
            FROM Symptoms symptoms6_
            WHERE symptoms6_.id=case0_.symptoms_id ) IS NULL)
     AND case0_.reportDate>at_end_of_day(CASE
                                             WHEN vaccinatio4_.vaccinationDate IS NULL THEN timestamp_subtract_14_days(vaccinatio4_.reportDate)
                                             ELSE vaccinatio4_.vaccinationDate
                                         END))
LEFT OUTER JOIN immunization immunizati7_ ON vaccinatio4_.immunization_id=immunizati7_.id
LEFT OUTER JOIN Region region8_ ON case0_.region_id=region8_.id
LEFT OUTER JOIN District district9_ ON case0_.district_id=district9_.id
LEFT OUTER JOIN Facility facility10_ ON case0_.healthFacility_id=facility10_.id
LEFT OUTER JOIN PointOfEntry pointofent11_ ON case0_.pointOfEntry_id=pointofent11_.id
LEFT OUTER JOIN users user12_ ON case0_.surveillanceOfficer_id=user12_.id
LEFT OUTER JOIN Region region13_ ON case0_.responsibleRegion_id=region13_.id
LEFT OUTER JOIN District district14_ ON case0_.responsibleDistrict_id=district14_.id
LEFT OUTER JOIN users user15_ ON case0_.reportingUser_id=user15_.id
LEFT OUTER JOIN Hospitalization hospitaliz16_ ON case0_.hospitalization_id=hospitaliz16_.id
LEFT OUTER JOIN previoushospitalization previousho17_ ON hospitaliz16_.id=previousho17_.hospitalization_id
LEFT OUTER JOIN ClinicalCourse clinicalco18_ ON case0_.clinicalCourse_id=clinicalco18_.id
LEFT OUTER JOIN Symptoms symptoms19_ ON case0_.symptoms_id=symptoms19_.id
LEFT OUTER JOIN Therapy therapy20_ ON case0_.therapy_id=therapy20_.id
LEFT OUTER JOIN HealthConditions healthcond21_ ON case0_.healthConditions_id=healthcond21_.id
LEFT OUTER JOIN MaternalHistory maternalhi22_ ON case0_.maternalHistory_id=maternalhi22_.id
LEFT OUTER JOIN PortHealthInfo porthealth23_ ON case0_.portHealthInfo_id=porthealth23_.id
LEFT OUTER JOIN sormastosormasorigininfo sormastoso24_ ON case0_.sormasToSormasOriginInfo_id=sormastoso24_.id
LEFT OUTER JOIN sormastosormasshareinfo sormastoso25_ ON case0_.id=sormastoso25_.caze_id
LEFT OUTER JOIN EpiData epidata26_ ON case0_.epiData_id=epidata26_.id
LEFT OUTER JOIN exposures exposures27_ ON epidata26_.id=exposures27_.epiData_id
LEFT OUTER JOIN LOCATION location28_ ON exposures27_.location_id=location28_.id
LEFT OUTER JOIN activityascase activities29_ ON epidata26_.id=activities29_.epiData_id
LEFT OUTER JOIN LOCATION location30_ ON activities29_.location_id=location30_.id
LEFT OUTER JOIN samples samples31_ ON case0_.id=samples31_.associatedCase_id
LEFT OUTER JOIN PathogenTest pathogente32_ ON samples31_.id=pathogente32_.sample_id
LEFT OUTER JOIN Person person33_ ON case0_.person_id=person33_.id
LEFT OUTER JOIN LOCATION location34_ ON person33_.address_id=location34_.id
LEFT OUTER JOIN Visit visits35_ ON case0_.id=visits35_.caze_id
LEFT OUTER JOIN surveillancereports surveillan36_ ON case0_.id=surveillan36_.caze_id
WHERE case0_.deleted=?
  AND (EXISTS
         (SELECT case39_.id
          FROM externalshareinfo externalsh38_
          LEFT OUTER JOIN cases case39_ ON externalsh38_.caze_id=case39_.id
          WHERE case39_.id=case0_.id
          GROUP BY case39_.id
          HAVING case0_.changeDate>max(externalsh38_.creationDate)
          AND (case0_.changeDate IS NOT NULL)
          OR symptoms19_.changeDate>max(externalsh38_.creationDate)
          AND (symptoms19_.changeDate IS NOT NULL)
          OR hospitaliz16_.changeDate>max(externalsh38_.creationDate)
          AND (hospitaliz16_.changeDate IS NOT NULL)
          OR previousho17_.changeDate>max(externalsh38_.creationDate)
          AND (previousho17_.changeDate IS NOT NULL)
          OR therapy20_.changeDate>max(externalsh38_.creationDate)
          AND (therapy20_.changeDate IS NOT NULL)
          OR clinicalco18_.changeDate>max(externalsh38_.creationDate)
          AND (clinicalco18_.changeDate IS NOT NULL)
          OR healthcond21_.changeDate>max(externalsh38_.creationDate)
          AND (healthcond21_.changeDate IS NOT NULL)
          OR maternalhi22_.changeDate>max(externalsh38_.creationDate)
          AND (maternalhi22_.changeDate IS NOT NULL)
          OR porthealth23_.changeDate>max(externalsh38_.creationDate)
          AND (porthealth23_.changeDate IS NOT NULL)
          OR sormastoso24_.changeDate>max(externalsh38_.creationDate)
          AND (sormastoso24_.changeDate IS NOT NULL)
          OR sormastoso25_.changeDate>max(externalsh38_.creationDate)
          AND (sormastoso25_.changeDate IS NOT NULL)
          OR epidata26_.changeDate>max(externalsh38_.creationDate)
          AND (epidata26_.changeDate IS NOT NULL)
          OR exposures27_.changeDate>max(externalsh38_.creationDate)
          AND (exposures27_.changeDate IS NOT NULL)
          OR location28_.changeDate>max(externalsh38_.creationDate)
          AND (location28_.changeDate IS NOT NULL)
          OR activities29_.changeDate>max(externalsh38_.creationDate)
          AND (activities29_.changeDate IS NOT NULL)
          OR location30_.changeDate>max(externalsh38_.creationDate)
          AND (location30_.changeDate IS NOT NULL)
          OR samples31_.changeDate>max(externalsh38_.creationDate)
          AND (samples31_.changeDate IS NOT NULL)
          OR pathogente32_.changeDate>max(externalsh38_.creationDate)
          AND (pathogente32_.changeDate IS NOT NULL)
          OR person33_.changeDate>max(externalsh38_.creationDate)
          AND (person33_.changeDate IS NOT NULL)
          OR location34_.changeDate>max(externalsh38_.creationDate)
          AND (location34_.changeDate IS NOT NULL)
          OR visits35_.changeDate>max(externalsh38_.creationDate)
          AND (visits35_.changeDate IS NOT NULL)
          OR surveillan36_.changeDate>max(externalsh38_.creationDate)
          AND (surveillan36_.changeDate IS NOT NULL)
          OR vaccinatio4_.changeDate>max(externalsh38_.creationDate)
          AND (vaccinatio4_.changeDate IS NOT NULL)
          OR immunizati7_.changeDate>max(externalsh38_.creationDate)
          AND (immunizati7_.changeDate IS NOT NULL)))
  AND case0_.dontShareWithReportingTool=FALSE
ORDER BY greatest(case0_.changeDate, person1_.changeDate) DESC
LIMIT ?
```

</details>

**Review by StefanKock — APPROVED** [review]

**Pull request merged**